arbnode/batch_poster.go

```diff
@@ -1037,7 +1045,13 @@ const ethPosBlockTime = 12 * time.Second
 
 var errAttemptLockFailed = errors.New("failed to acquire lock; either another batch poster posted a batch or this node fell behind")
 
-func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error) {
+func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (ret bool, err error) {
+	defer func() {
+		if err != nil {
+			batchPosterFailureCounter.Inc(1)
```

Comment:
Could we just put this metric at the callsite instead of naming the return values? Naming the return values is a bit confusing if unnecessary.

Reply:
Main reason for doing it this way is you avoid having to do an Inc(1) at each error site, which is a lot in this function. It also means if you add a new error path, you don't have to remember to add the Inc(1). If that's preferred though I can do it that way

Reply:
I meant at the callsite of maybePostBatch (IIRC there's only one)

Reply:
Ah I put it at the top in case any intermediate functions error out. That would still prevent batch posting if I'm reading it correctly and be something we'd want to show up in the error metric.

Reply:
I meant putting it here https://github.com/OffchainLabs/nitro/blob/a45842cddc7f19bbcfe6c05bfd1b2f797584f8fc/arbnode/batch_poster.go#L1498 wouldn't that be the same as what you have with the defer?

Reply:
Oh yes you're right, I'll make the update